src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs

```diff
@@ -1718,5 +1718,96 @@ private static bool TryFindFirstMatchedLane(Vector128<ushort> compareResult, ref
         }
+
+        public static void Reverse(ref char buf, nuint length)
+        {
+            nint numBytes = (int)length * sizeof(char);
+            int numBytesWritten = 0;
+            ref byte first = ref Unsafe.As<char, byte>(ref buf);
+            ref byte last = ref Unsafe.Add(ref Unsafe.Add(ref first, numBytes), -sizeof(char));
+            if (Avx2.IsSupported && Vector256<byte>.Count * 2 <= numBytes)
+            {
+                Vector256<byte> ReverseMask = Vector256.Create(
+                    (byte)14, 15, 12, 13, 10, 11, 8, 9, 6, 7, 4, 5, 2, 3, 0, 1, // first 128-bit lane
+                    14, 15, 12, 13, 10, 11, 8, 9, 6, 7, 4, 5, 2, 3, 0, 1); // second 128-bit lane
+                last = ref Unsafe.Add(ref Unsafe.Add(ref first, numBytes), -Vector256<byte>.Count);
+                do
+                {
+                    // Load in values from beginning and end of the array.
+                    Vector256<byte> tempFirst = Unsafe.As<byte, Vector256<byte>>(ref first);
+                    Vector256<byte> tempLast = Unsafe.As<byte, Vector256<byte>>(ref last);
+
+                    // Avx2 operates on two 128-bit lanes rather than the full 256-bit vector.
+                    // Perform a shuffle to reverse each 128-bit lane, then permute to finish reversing the vector:
+                    //     +---------------------------------------------------------------+
+                    //     | A | B | C | D | E | F | G | H | I | J | K | L | M | N | O | P |
+                    //     +---------------------------------------------------------------+
+                    //         Shuffle --->
+                    //     +---------------------------------------------------------------+
+                    //     | H | G | F | E | D | C | B | A | P | O | N | M | L | K | J | I |
+                    //     +---------------------------------------------------------------+
+                    //         Permute --->
+                    //     +---------------------------------------------------------------+
+                    //     | P | O | N | M | L | K | J | I | H | G | F | E | D | C | B | A |
+                    //     +---------------------------------------------------------------+
+                    tempFirst = Avx2.Shuffle(tempFirst, ReverseMask);
+                    tempFirst = Avx2.Permute2x128(tempFirst, tempFirst, 1);
+                    tempLast = Avx2.Shuffle(tempLast, ReverseMask);
+                    tempLast = Avx2.Permute2x128(tempLast, tempLast, 1);
+
+                    // Store the reversed vectors
+                    Unsafe.As<byte, Vector256<byte>>(ref first) = tempLast;
+                    Unsafe.As<byte, Vector256<byte>>(ref last) = tempFirst;
+
+                    // Adjust the references to point to the next vector
+                    first = ref Unsafe.Add(ref first, Vector256<byte>.Count);
+                    last = ref Unsafe.Add(ref last, -Vector256<byte>.Count);
```

Comment:
Semantically, using [Subtract](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.unsafe.subtract?view=net-6.0) would be better.
```suggestion
                    last = ref Unsafe.Subtract(ref last, Vector256<byte>.Count);
```